src/libtego/source/ed25519.cpp

```diff
@@ -206,6 +206,21 @@ extern "C"
         }, error);
     }
 
+    tego_bool_t tego_v3_onion_service_id_string_is_valid(
+        const char* serviceIdString,
+        size_t serviceIdStringLength,
+        tego_error_t** error)
+    {
+        return tego::translateExceptions([&]() -> tego_bool_t
+        {
+            TEGO_THROW_IF_NULL(serviceIdString);
+
+            auto serviceId = std::make_unique<tego_v3_onion_service_id>(serviceIdString, serviceIdStringLength);
+            // if the constructor returns without throwing an error, then it's a valid id
```

Comment:
So this works, but I've some nit-picking to do. First off, we don't need to do a make_unique here. We can avoid the heap allocation by just doing:
```c++
tego_v3_onion_service_id serviceId(serviceIdString, serviceIdStringLength);
```

That said, what we should be doing here is ripping out the relevant code from the tego_v3_onion_service_id constructor into a static method:

```c++
struct tego_v3_onion_service_id {
...
    static bool is_valid(std::string_view serviceIdString);
};
```
And calling that from within both the constructor and our new `tego_v3_onion_service_id_string_is_valid(...)` function. This will avoid the unnecessary copy at the end of the current constructor.